ui/components/multichain/asset-picker-amount/asset-picker-modal/AssetList.tsx

```diff
@@ -37,8 +50,21 @@ export default function AssetList({
 }: AssetListProps) {
   const selectedToken = asset.details?.address;
 
-  const nativeCurrency = useSelector(getNativeCurrency);
-  const balanceValue = useSelector(getSelectedAccountCachedBalance);
+  const nativeCurrency = useSelector(getMultichainNativeCurrency);
+  const { chainId, ticker, type } = useSelector(getMultichainCurrentNetwork);
+  const { rpcUrl } = useSelector(getProviderConfig);
+
+  const showFiat = useSelector(getMultichainShouldShowFiat);
+  const { useNativeCurrencyAsPrimaryCurrency } = useSelector(getPreferences);
+
+  const isOriginalNativeSymbol = useIsOriginalNativeTokenSymbol(
+    chainId,
+    ticker,
+    type,
+    rpcUrl,
+  );
+
+  const selectedAccountBalance = useSelector(getSelectedAccountCachedBalance);
 
```

Comment:
Let's avoid adding new logic in a hotfix PR. Feel free to make a follow up and I can review but we initially avoided this logic altogether since it blew up that network's RPC endpoint.

Reply:
This additional logic enables a more effective resolution of the mentioned bugs without leading to regressions in other areas, which is integral to the solution (since alterations in the wallet view typically affect the asset picker). An alternative approach could have involved completely separating the asset picker and the token list wallet view to facilitate such fixes.

Reply:
The bug this addresses (i.e. not showing the user's currency code for the native asset when the user's primary currency is set to fiat) is largely unrelated to the styling regressions mentioned in #25575. While we reexpose this bug by showing the symbol again, this edge case always existed and is not part of the regressions this PR is targeting.

This is something we should address, the existing solution was known to cause a cascade of bugs (see: #24859). While #25501 aims to prevent the rate limiting that caused these bugs, I wouldn't be comfortable reintroducing the hook until we're sure it's no longer problematic. At the very least, **this should be in a separate PR.**